### ISSUE TITLE
Can't enter NORMAL mode with Vim extension when the plugins enabled

### ISSUE THREAD

**Issue by crosstyan** [emilast/vscode-logfile-highlighter#588]
Type: <b>Bug</b>

1. Install [vim entesnion](https://marketplace.visualstudio.com/items?itemName=vscodevim.vim)
2. Install [Log FIle Highlighter](https://marketplace.visualstudio.com/items?itemName=emilast.LogFileHighlighter)
3. Open any file
4. You're stuck at INSERT mode

Extension version: 3.0.0
VS Code version: Code 1.90.1 (Universal) (611f9bfce64f25108829dd295f54a6894e87339d, 2024-06-11T21:02:41.372Z)
OS version: Darwin arm64 23.5.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 (8 x 2400)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|14, 9, 10|
|Memory (System)|8.00GB (0.42GB free)|
|Process Argv|--crash-reporter-id 25997ac5-c3e6-474f-92ba-e17c30625f32|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492:30256859
vscorecescf:30445987
vscod805cf:30301675
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythonregdiag2:30936856
pythonmypyd1:30879173
2e7ec940:31000449
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
jchc7451:31067544
chatpanelt:31048053
dsvsc021:30996838
0ee40948:31013168
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
dwnewjupytercf:31046870
legacy_priority:31077528

```

</details>

**Comment by moliyadi** [comment]
+1

**Comment by VictorPerezForm3** [comment]
Yep, This is definitely happening

**Comment by emilast** [comment]
Ok, this sounds bad. Can someone explain what you do to enter NORMAL mode. Is the ESC key involved?

**Comment by emilast** [comment]
I just released 3.0.1 with a change to the Escape key binding so that it only applies in log files rather than globally (which was a mistake in 3.0.0). Does this improve things?

**Comment by crosstyan** [comment]
I guess it's fixed in the latest version, thanks!

**Comment by emilast** [comment]
@crosstyan, I have tried to repeat the remaining problem but need help with understanding the Vim extension. If I understand it correctly, when that extension is installed, pressing ESC in a file should switch modes? That doesn't happen after I install the extension, regardless of whether LogFileHighlighter is installed or not, so I assume I'm doing something wrong. If I cannot repeat the problem, it will be difficult to correct it.

**Comment by crosstyan** [comment]
> when that extension is installed, pressing ESC in a file should switch modes?

No, pressing ESC will exit INSERT mode and back to NORMAL mode. In my case I was opening the file with INSERT mode by default, which might not be the default VIM behavior. 
To enter INSERT mode from NORMAL mode you need to press `I` first, and then back to NORMAL mode by pressing ESC. You could check which mode you're in by the Status Bar at the bottom of VSCode UI.

**Comment by crosstyan** [comment]
https://github.com/emilast/vscode-logfile-highlighter/assets/49297668/c0e6f140-bd5d-4109-b908-fe8366b5aff4

Here's a short video I record for reproducing the problem.

**Comment by emilast** [comment]
@crosstyan, I believe this should now be fixed in 3.0.3 that I just released. I added an extra condition to the key binding so it only applies if there are rendered timestamp progress indicators. If there are no indicators to clear, the Escape key event should be propagated to the Vim extension and everything should work as before.

It would be great if you could confirm the fix after the extension has been updated, and please remember to **restart VS Code** an extra time before testing so that the key bindings are reset properly after the update.

Thanks in advance!

**Comment by crosstyan** [comment]
fix confirmed, thanks
